src/mon/OSDMonitor.cc

```diff
@@ -794,6 +805,12 @@ void OSDMonitor::create_pending()
 	       << " -> " << pg_map->nearfull_ratio << " (from pgmap)" << dendl;
       pending_inc.new_nearfull_ratio = pg_map->nearfull_ratio;
     }
+    // transition lec from pgmap to min_last_epoch_clean_by_pg
+    min_last_epoch_clean_by_pg.clear();
+    for (const auto& pg_stat : pg_map->pg_stat) {
+      auto lec = pg_stat.second.get_effective_last_epoch_clean();
+      min_last_epoch_clean_by_pg[pg_stat.first] = lec;
+    }
```

Comment:
i just want to make sure we have all PGs in `min_last_epoch_clean_by_pg` before using it to trim osdmap. otherwise, the value of min_lec could be determined by a subset of pg, that would be catastrophic.